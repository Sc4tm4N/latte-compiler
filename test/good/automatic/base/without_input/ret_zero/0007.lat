// print int variables according to int max and int min (compared to C)
int main() {
  int INT_MIN = -2147483648;
  int INT_MAX = 2147483647;
  printInt(INT_MIN + INT_MAX);
  printInt(INT_MIN - INT_MAX);
  printInt(INT_MIN * INT_MAX);
  printInt(INT_MIN / INT_MAX);
  printInt(INT_MIN % INT_MAX);
  printInt(INT_MAX + INT_MIN);
  printInt(INT_MAX - INT_MIN);
  printInt(INT_MAX * INT_MIN);
  printInt(INT_MAX / INT_MIN);
  printInt(INT_MAX % INT_MIN);
  return 0;
}