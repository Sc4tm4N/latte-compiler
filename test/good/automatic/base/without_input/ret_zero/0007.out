-1
1
-2147483648
-1
-1
-1
-1
-2147483648
0
2147483647
